yosys_pass.py

```diff
@@ -2,11 +2,45 @@
 import re
 import subprocess
 from typing import Dict, List
+from tempfile import mkstemp
+from shutil import move, copymode
 
 from checker import extract_inputs_outputs, extract_module_signature
 
 
+
+def fix_module_name(input_path: str):
+    file = input_path.split('/')[-1]
+    subst = file[:file.rfind('.')]  # get pure name
+    fh, abs_path = mkstemp()
+    with os.fdopen(fh, 'w') as new_file:
+        with open(input_path) as old_file:
+            for line in old_file:
+                if 'module' in line and 'endmodule' not in line:
+
+                    # This pattern divides the module declaration into three parts.
+                    # for example: it divides this: module \nulls9-ko23iou09vn2adder(pi0, pi1, pi2, pi3, po0, po1, po2);\n
+                    # into this:
+                    # match.group(1) <= 'module '
+                    # match.group(2) <= '\nulls9-ko23iou09vn2adder'
+                    # match.group(1) <= '(pi0, pi1, pi2, pi3, po0, po1, po2);\n'
```

Comment:
Just a quick remark, the third match group should be match.group(3) and the resulting string should not contain the new line character.

Reply:
Thanks! You are correct! 
Good thing that we don't use the third group! :D